proto/mls/message_contents/group_metadata.proto

```diff
@@ -0,0 +1,63 @@
+// Group metadata
+syntax = "proto3";
+
+package xmtp.mls.message_contents;
+
+option go_package = "github.com/xmtp/proto/v3/go/mls/message_contents";
+option java_package = "org.xmtp.proto.mls.message.contents";
+
+// Parent message for group metadata
+message GroupMetadataV1 {
+  ConversationType conversation_type = 1;
+  string creator_account_address = 2;
+  PolicySet policies = 3;
+}
+
+// Defines the type of conversation
+enum ConversationType {
+  CONVERSATION_TYPE_UNSPECIFIED = 0;
+  CONVERSATION_TYPE_GROUP = 1;
+  CONVERSATION_TYPE_DM = 2;
+}
+
+// The set of policies that govern the group
+message PolicySet {
+  MembershipPolicy add_member_policy = 1;
+  MembershipPolicy remove_member_policy = 2;
+  MembershipPolicy add_installation_policy = 3;
+  MembershipPolicy remove_installation_policy = 4;
+}
+
+// A policy that governs adding/removing members or installations
+message MembershipPolicy {
+  // Base policy
+  enum BasePolicy {
+    BASE_POLICY_UNSPECIFIED = 0;
+    BASE_POLICY_ALLOW = 1;
+    BASE_POLICY_DENY = 2;
+    BASE_POLICY_ALLOW_SAME_MEMBER = 3;
```

Comment:
This would only be used for removing members right? Is there any case where it would be permissible for this to NOT be set for removing members?

Reply:
Probably not. Think that one would be pretty uniform.

Reply:
I think having this specified as a policy, rather than baked directly into the code, means a malicious person could potentially create a group with this option turned off (no self removal allowed)? I guess my instinct with these changes is to expose the minimum amount for developers to have the features they desire, rather than having maximum flexibility